testbed/testbed/utils/rawLibp2p.go

```diff
@@ -0,0 +1,95 @@
+package utils
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/ipfs/go-cid"
+	files "github.com/ipfs/go-ipfs-files"
+	ipld "github.com/ipfs/go-ipld-format"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+)
+
+type RawLibp2pNode struct {
+	h host.Host
+}
+
+func CreateRawLibp2pNode(ctx context.Context, h host.Host, nodeTP NodeType) (*RawLibp2pNode, error) {
```

Comment:
What about merging this code with the one for the HTTPlibp2pNode so we have a single libp2pNode definition where we register handlers, and an attribute in the libp2p node that defines the type of exchange to perform? That way we can easily accommodate new protocols over the libp2p node without a lot of code duplication. What do you think?

Reply:
honestly they look quite a bit different. I feel like part of the node interface's point it to be the container for more protocol tests --I'm personally ok with these being two different node implementations.

Reply:
but I also see your point.

Reply:
Going ahead with different files for now as the implementation does differ considerably.

Reply:
After a second review, I am convinced :) Go for it!